### PR TITLE
coqPackages.mathcomp-analysis: etc

### PR DESCRIPTION
**Pull request by CohenCyril**
###### Description of changes
hb dep is added later, after we check the version number

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/coq-modules/mathcomp-analysis/default.nix [hunk 1]
```diff
@@ -39,7 +38,7 @@ let
   packages = [ "classical" "analysis" ];
 
   mathcomp_ = package: let
-      classical-deps = [ mathcomp.algebra mathcomp-finmap hierarchy-builder ];
+      classical-deps = [ mathcomp.algebra mathcomp-finmap ];
```
Comment (proux01) [inline pkgs/development/coq-modules/mathcomp-analysis/default.nix:41]: @CohenCyril I'm not sure I get that change, it seems to me that old (pre MathComp 2) versions of analysis still depend on hierarchy-builder?

Comment (CohenCyril) [reply pkgs/development/coq-modules/mathcomp-analysis/default.nix:41]: The dependency is already [added below](https://github.com/NixOS/nixpkgs/pull/237928/files#diff-0bdd8a94e555cca69d7d4900f59a73697c7b083d531c33e865feefe3a5605b13R84). Adding it unconditionally breaks mathcomp < 0.3.4

Comment (proux01) [reply pkgs/development/coq-modules/mathcomp-analysis/default.nix:41]: Indeed, thanks for the explanation.

**Comment by proux01** [comment]
@vbgl successfully tested on https://github.com/coq-community/coq-nix-toolbox/pull/150

**Pull request merged**